Week8/app/src/main/java/com/bartoszkrol/catfacts/App.kt

```diff
@@ -17,6 +21,31 @@ class App : Application() {
     override fun onCreate() {
         super.onCreate()
         instance = this
+
+        startDownloadingDataWorker()
+    }
+
+    /**
+     * Starts a worker that downloads a data from the API server every 1 hour.
+     * This is a unique worker, so there will be only 1 instance of it.
+     */
+    private fun startDownloadingDataWorker() {
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .setRequiresBatteryNotLow(true)
+            .setRequiresStorageNotLow(true)
+            .build()
+
+        val work = PeriodicWorkRequestBuilder<DownloadDataWorker>(1, TimeUnit.MINUTES)
+            .setConstraints(constraints)
+            .build()
+
+        val workManager = WorkManager.getInstance(this)
```

Comment:
Really nice and well structured! :]

However, as someone (may be you) mentioned, the minimum period is 15 minutes :/

But I'll take it as if it's working, so don't worry.

Reply:
Yes, the minimum period is 15minutes (I think it apply for JobScheduler too).

I forgot to change repeatInterval value and timeUnit. I was testing it on the real device to see if its gonna work every 1 minute.